### PR TITLE
Fix prefetching of edit, children and select links

### PR DESCRIPTION
**Pull request by aschempp**
Replaces https://github.com/contao/contao/pull/7423

I think this is a much safer option than #7423 because we do not try to detect operations, but actually mark the ones we want to prefetch. It also allows others to mark their operations (e.g. see theme modules) as to be prefetched (and therefore not have a request token).

Btw. if the request token would be omitted accidentally on an operation, the warning dialog would be triggered, so that should still be safe.

The new Stimulus controllers fixes the issue of preventing Turbo on the operations due to calling `stopPropagation()` on all links in `Theme.setupCtrlClick`.

### CODE REVIEW

**Review by fritzmg — COMMENTED** [review]
Having the ability for `prefetch => true` is good - but controlling this then via the request token seems odd to me.

And regardless of Turbo I still think we should remove the `rt=` parameter from all operations that do not actually require one.

**Comment by aschempp** [comment]
In my opinion, an operation can only be prefetched if it does not require a request token to execute. So they are synonymous "enabling prefetch" seems like a feature, but "disable request token" seems like an implementation detail. I also intentionally did not include `turbo` in the property name because it is unrelated.

Another option I can think of is marking this as "safe method". But ultimately, I think we should have all operations prefetch automatically that we can, except if they have `prefetch=false` (which is the whole point here).

File: core-bundle/assets/controllers/deeplink-controller.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        this.element.addEventListener('click', this.visitSecondary);
+    }
+
+    disconnect () {
+        this.element.removeEventListener('click', this.visitPrimary);
+        this.element.removeEventListener('click', this.visitSecondary);
+    }
+
+    visitPrimary (event) {
+        const key = window.navigator.platform?.startsWith('Mac') ? 'metaKey' : 'ctrlKey';
```
Comment (aschempp) [inline core-bundle/assets/controllers/deeplink-controller.js]: I would prefer to use native Stimulus events here (`data-action="click.alt->contao--deeplink#visitPrimary`), but unfortunately there is no way to have a different key modifier per platform. Implementing our own key seems kinda overkill too, of would we have any place to reuse that?

Comment (fritzmg) [reply core-bundle/assets/controllers/deeplink-controller.js]: Turbo seems to support

```html
data-action="alt+click->contao--deplink#visitPrimary"
```

which will automatically be the option key on MacOS.

https://github.com/hotwired/stimulus/pull/665
https://stimulus.hotwired.dev/reference/actions#keyboardevent-filter

Comment (aschempp) [reply core-bundle/assets/controllers/deeplink-controller.js]: yes, but we are using meta key on mac and ctrl key on windows

Comment (leofeyer) [reply core-bundle/assets/controllers/deeplink-controller.js]: Maybe we can unify that in a separate PR?

Comment (aschempp) [reply core-bundle/assets/controllers/deeplink-controller.js]: What do you mean by "unify"?

Comment (leofeyer) [reply core-bundle/assets/controllers/deeplink-controller.js]: Unify == use the same key combination on all operating systems.

Comment (aschempp) [reply core-bundle/assets/controllers/deeplink-controller.js]: Sounds good to me. Or generally rethink some "quick edit" stuff when reworking the operations (context menu etc.). I'd leave this as it is here then.

**Comment by leofeyer** [comment]
As discussed in the Contao call, we want to merge this pull request instead of #7423, but only in version 5.5.

**Review by fritzmg — DISMISSED** [review]
I still think we should use `data-turbo-prefetch="false"` wherever we do not want to prefetch, instead of relying on the `rt` query parameter.

But approving in any case.

**Review by m-vo — DISMISSED** [review]

**Review by leofeyer — CHANGES_REQUESTED** [review]
There is a behavior change that needs to be fixed:

Before this change, `[Cmd]` + click would open the primary target and `[Cmd]` + `[Shift]` + click would open the secondary. However, now `[Cmd]` + `[Shift]` + click also opens the primary target and just `[Shift]` + click opens the secondary (and also selects half of the content).

Can you please fix this so the script behaves like it used to?

**Comment by aschempp** [comment]
Fixed in 09dd3ee704f5fe7bcda2542a980429ef1c0e74e7

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**